chains/heads/tracker.go

```diff
@@ -264,6 +299,18 @@ func (t *tracker[HTH, S, ID, BLOCK_HASH]) handleNewHead(ctx context.Context, hea
 			promOldHead.WithLabelValues(t.chainID.String()).Inc()
 			err := fmt.Errorf("got very old block with number %d (highest seen was %d)", head.BlockNumber(), prevHead.BlockNumber())
 			t.log.Critical("Got very old block. Either a very deep re-org occurred, one of the RPC nodes has gotten far out of sync, or the chain went backwards in block numbers. This node may not function correctly without manual intervention.", "err", err)
+
+			var finalityErr error
+			if prevLatestFinalized.BlockNumber() == head.BlockNumber() {
```

Comment:
Could you add a test that covers handleOnNewHead too?
Also, it seems like we need to do these checks before saving to the storage, otherwise, on the next iteration, we won't observe the issue even if chain's state is still inconsistent